server/lib/nl/fulfillment/comparison.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from lib.nl.nl_detection import ClassificationType, Detection, \
+  ContainedInPlaceType, ContainedInClassificationAttributes, \
+    RankingType, RankingClassificationAttributes, ComparisonClassificationAttributes, \
+    Place, ClassificationAttributes
+
+from lib.nl.nl_utterance import Utterance, ChartOriginType, ChartSpec, \
+  ChartType, CTX_LOOKBACK_LIMIT
+
+from lib.nl.fulfillment.base import populate_charts_for_places, PopulateState, ChartVars, \
+  add_chart_to_utterance
+
+from lib.nl.fulfillment import context
+
+
+def populate(uttr: Utterance) -> bool:
+  # NOTE: The COMPARISON attribute has no additional parameters.  So start
+  # by directly inferring the list of places to compare.
+  state = PopulateState(uttr=uttr,
+                        main_cb=_populate_comparison_cb,
+                        fallback_cb=_fallback_comparison_cb)
+  for places_to_compare in context.places_for_comparison_from_context(uttr):
+    if populate_charts_for_places(state, places_to_compare):
+      return True
+  return False
+
+
+def _populate_comparison_cb(state: PopulateState, chart_vars: ChartVars,
```

Comment:
super nit: can these just be _populate_cb and _fallback_cb everywhere?

Reply:
good idea!